File: TD/TD4/Code/4.cpp
/**
 * 1. Fonction -> Retourner le perimetre d'un cercle
 * 2. Fonction -> Retourner l'aire d'un cercle
 * 3. Sous-programme -> Utliser ces deux fonction
 */
#include <iostream>

/* Fonction 1 */

float perimCercle(int ray)
{
  float pi;
  float per;
  
  pi = 3.14;
  per = 2 * pi * ray;
  
  return per;
}

/* Fonction 2 */

float aireCercle(int ray)
{
  float pi;
  float air;
  
  pi = 3.14;
  air = pi * ray * ray;
  
  return air;
}

/* Procédure */

void sousProg(int ray, float &per, float &air)
{
  per = perimCercle(ray);
  air = aireCercle(ray);
}

/* Utilisation */

int main()
{
  int ray;
  float per;
  float air;
  
  std::cout << "Donne moi une valeur : " << std::flush;
  std::cin >> ray;
  
  sousProg(ray, per, air);
  
  std::cout << "Le perimetre est " << per << std::endl;
  std::cout << "La aire est " << air << std::endl;
  
  return 0;
}

File: TD/TD4/Code/Python/4.py
#!/ust/bin/python
# 1. Retourner le perimetre d'un cercle
# 2. Retourner l'aire d'un cercle
# 3. Utliser ces deux fonction
from math import pi


def perim_cercle(ray):
    per = int(2 * pi * ray)
    return per


def aire_cercle(ray):
    aire = int(pi * ray ** 2)
    return aire


def perim_et_aire(ray):
    per = perim_cercle(ray)
    aire = aire_cercle(ray)
    return per, aire


print('Donne moi une valeur : ', end='')
foo = int(input())

per, aire = perim_et_aire(foo)

print('Le périmètre de ton cercle est :', per,
      'et l\' aire de ce cercle est', aire)
